src/taker/config.rs

```diff
@@ -89,13 +94,22 @@ impl TakerConfig {
         })
     }
 
-    // Method to manually serialize the Taker Config into a TOML string
+    /// Writes the current configuration to a TOML file at the specified path.
+    ///
+    /// This method serializes the TakerConfig into a TOML format and writes it to disk.
+    /// It creates the parent directory if it doesn't exist.
     pub(crate) fn write_to_file(&self, path: &Path) -> std::io::Result<()> {
         let toml_data = format!(
-            "control_port = {}
+            "# Taker Configuration File
+# Control port for Tor control interface
+control_port = {}
+# Socks port for Tor proxy
 socks_port = {}
+# Authentication password for Tor control interface
 tor_auth_password = {}
+# Directory server address (can be clearnet or onion)
 directory_server_address = {}
+# Connection type (TOR or CLEARNET)
```

Comment:
Same thing we can attempt to remove connection type from config in a followup PR.